### ISSUE TITLE
Simple Echobot gives serviceURL null error, when called from slack, after update to v3

### ISSUE THREAD

**Issue by niyatia** [Microsoft/BotBuilder#805]
Hi,

I upgraded to v3 of the bot framework and emulator and have created a simple echo bot as in the samples and successfully ran it using the emulator. However, when I integrated it with slack, it gives Activity.serviceURL null error when messaging the bot. I get the http://localhost:9000 as serviceURL in the Activity when I run it in emulator but can't get any url when running it from slack. It worked perfectly fine before the upgrade on both emulator and slack. Any suggestions on how to get my bot running in slack after upgrade?

Thanks
Niyati

**Comment by willportnoy** [comment]
Is this the Node or C# SDK?  There is an issue where the ServiceUrl is not copied to the new message when CreateReply is used.

**Comment by msft-shahins** [comment]
@niyatia what is your bot's Microsoft App Id? Incoming activities from channels should have valid ServiceUrl. Are you seeing the same behavior with other channels?

**Comment by niyatia** [comment]
@msft-shahins  I did not try with other channels. I can try integrating with skype. My appId is Oktabot. Also I am using C# sdk @willportnoy .

**Comment by niyatia** [comment]
Never mind. I realized I did not update the settings for my bot under botframework->MyBots->EchoBot->Edit and change the version from 1.0 to 3.0. I had created this bot in 1.0 and published to slack. Later I updated the framework and made required changes in code and published again but missed this configuration.  After that update, I am able to connect the bot via slack. Thanks for the help. Closing the issue but I would say documentation about the upgrade is not very clear or helpful.
